### ISSUE TITLE
Automated release/delivery process based on Travis-CI

### ISSUE THREAD

**Issue by Paebbels** [mplanchard/pydecor#7]
In addition to #6, I would like to create a PR for automated deliveries from GitHub over Travis-CI to PyPI.

While normal pushes will trigger tests (Travis-CI + tox) and documentation creation (ReadTheDocs), deliveries to PyPI will only be triggered on tagged commits.

Steps:
1. modify `.travis-ci.yml`
2. Create a PyPI project limited access token
3. register token in Travis-CI as a variable
4. trigger pipeline for a test

Are you interested in such a CI/CD update?

**Comment by mplanchard** [comment]
Yep, I'm happy to move us towards automated deployments. I have a few scripts hanging around that make the process easier and/or safer. [This one](https://github.com/mplanchard/safetywrap/blob/master/scripts/check_ready_to_distribute.py) helps to ensure only master-branch tagged changes that match the package version are able to be distributed, for example, and [this one](https://github.com/pypiserver/pypiserver/blob/master/bin/bumpver.py) goes as far as checking to be sure the README is up-to-date.

Once we've got the updates to the CI file ready, I can pop in to PyPI, get a token, and add it to Travis!

**Comment by Paebbels** [comment]
Most of your first script's parts are not needed. This is als done in a declarative description in the CI config.

```yaml
language: python

install: skip

# if: tag IS present

script:
  - echo "No tests. Just deploying to PyPI."

deploy:
  provider: pypi
  user: "__token__"
  password: $PYPI_TOKEN
  distributions: "sdist bdist_wheel"
  skip_existing: true
  on:
    tags: true
```

This is a very simple Travis config:
- do nothing for build or test
- if pushed version was tagged, then deploy to PyPI
- the token is stored in a hidden variable (a.k.a secret)

Source: https://github.com/Paebbels/pyAttributes/blob/master/.travis.yml

In more complex variants, I let it build documentation (if not outsourced to ReadTheDocs) and collect test coverage, check dependencies and security issues as well as license scans. I can help to setup some of the tools, configs and scripts needed to become more automated.

**Comment by mplanchard** [comment]
Yep, I agree. Most of the first script is there b/c I generally just do manual deployment via a `make distribute` command

**Comment by mplanchard** [comment]
Resolved by #11
